### ISSUE TITLE
[DX] Date field settings: remove custom js and and use `#states` instead

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#6598]
This is one of many things I discovered while working on #6596 that I would like us to fix. Spliting it here, to avoid scope creep in the other issue.

At the top of `date.admin.js` there are a few lines of custom JS that is used to show/hide the timezone handling options if the `Hour` checkbox is not ticked in the granularity options. We should replace that with a `#states` implementation instead.

**Comment by klonos** [comment]
PR here: https://github.com/backdrop/backdrop/pull/4784

**Comment by argiepiano** [comment]
Thanks for this, @klonos. However, by removing the first part of the js file, you end up with "dead code" that is never invoked. I don't know if this (the fact that the rest of the functions will never be invoked) is problematic or not, as I haven't done actual testing, but this needs to be addressed. Either figure out if this is unnecessary and remove the dead code, or do something to invoke it.

**Comment by argiepiano** [comment]
For example, one problematic aspect of your PR is that, before your PR, when the timezone div is hidden by a click, the JS code sets the TZ dropdown to "none". Right now, with your approach, this will not happen. This may create issues upon form submission.

**Comment by klonos** [comment]
Right, right. It took me a few seconds and head-scratching to understand what you mean, but I see it now @argiepiano 👍🏼 ...let me revise the PR then. Will most likely need to remove the entire `date.admin.js` file, and handle the setting to `none` in the form validation/submission  I think.

**Comment by klonos** [comment]
> ...and handle the setting to `none` in the form validation/submission  I think.

It turns out that it was already "handled" in `date_field_settings_validate()` - but an error was thrown via `form_set_error()` instead.

New PR here: https://github.com/backdrop/backdrop/pull/4790 (I borked the previous PR when I renamed the branch in GitHub)

I've changed things so that the field value is set to `'none'` in the same scenario/conditions, so this basically retains the same functionality as before (when it comes to what is saved in config - don't need the error thrown and asking people to manually fix things after submitting the form, when we know what's wrong and how to fix it in the background for them).

So PR changes summary:
- removed the entire `date.admin.js` file (no "dead code")
- replaced the custom JS functionality with a `#states` implementation instead
- changed `date_field_settings_validate()` to remove the form error that was thrown when the hour granularity was removed and `tz_handling` was still set to something other than `'none'`
- instead of throwing that error, the value of `tz_handling` is set to `'none'` when the hour granularity is removed
- test slightly tweaked to cover all that + expanded to cover more
